primitives/account/src/logs.rs

```diff
@@ -0,0 +1,364 @@
+                        .as_ref()
+                        .map(|old_address| old_address == address)
+                        .unwrap_or(false)
+                    || new_validator_address
+                        .as_ref()
+                        .map(|new_address| new_address == address)
+                        .unwrap_or(false)
+            }
+            Log::DeleteValidator {
+                validator_address,
+                reward_address,
+            } => validator_address == address || reward_address == address,
+            Log::Unstake {
+                staker_address,
+                validator_address,
+                ..
+            } => {
+                staker_address == address
+                    || validator_address
+                        .as_ref()
+                        .map(|validator_address| validator_address == address)
+                        .unwrap_or(false)
+            }
+            Log::PayoutReward { to, .. } => to == address,
+            Log::Park {
+                validator_address, ..
+            } => validator_address == address,
+            Log::Slash {
+                validator_address, ..
+            } => validator_address == address,
+            Log::RevertContract { contract_address } => contract_address == address,
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+#[cfg_attr(feature = "serde-derive", derive(serde::Serialize, serde::Deserialize))]
+pub struct TransactionLog {
+    #[serde(rename = "hash")]
+    pub tx_hash: Blake2bHash,
+    pub logs: Vec<Log>,
+}
+
+impl TransactionLog {
+    pub fn new(tx_hash: Blake2bHash, logs: Vec<Log>) -> Self {
+        Self { tx_hash, logs }
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+#[cfg_attr(feature = "serde-derive", derive(serde::Serialize, serde::Deserialize))]
+// Renaming affects only the struct names and thus their tag, the "type" field.
+#[serde(rename_all = "kebab-case", tag = "type")]
+pub enum BlockLog {
+    #[serde(rename_all = "camelCase")]
+    AppliedBlock {
+        #[serde(rename = "inherents")]
+        inherent_logs: Vec<Log>,
+        block_hash: Blake2bHash,
+        block_number: u32,
+        timestamp: u64,
+        #[serde(rename = "transactions")]
+        tx_logs: Vec<TransactionLog>,
+    },
+
+    #[serde(rename_all = "camelCase")]
+    RevertedBlock {
+        #[serde(rename = "inherents")]
+        inherent_logs: Vec<Log>,
+        block_hash: Blake2bHash,
+        block_number: u32,
+        #[serde(rename = "transactions")]
+        tx_logs: Vec<TransactionLog>,
+    },
+}
+
+impl BlockLog {
+    pub fn is_revert_block_log(&self) -> bool {
+        match self {
+            BlockLog::AppliedBlock { .. } => false,
+            BlockLog::RevertedBlock { .. } => true,
+        }
+    }
+}
+
+// Account Info is used as a return type of transactions to the blockchain and stores the logs and receipt associated with the transaction
+#[derive(Debug, PartialEq, Eq)]
```

Comment:
Same here, aren't we missing `#[serde(rename_all = "camelCase")]`?

Reply:
No field would be affected by this rule, plus this struct is never sent as JSON AFAIK.